### ISSUE TITLE
Desktop: icon not displaying correctly and not able to lock to launcher

### ISSUE THREAD

**Issue by matsest** [tutao/tutanota#981]
Great work on the desktop client! A little hickup from my initial testing:

### Client

Desktop: Linux (Ubuntu 18.04 with Unity)
Version: 3.44.9

#### Icon issue

There is an icon in ``~/.local/share/icons/hicolor/0x0/apps/appimagekit-tutanota-desktop.png``. Note that since this is in 0x0-directory, this is not recognized by my launcher. I have other icons from AppImage-applications in ``~/.local/share/icons/hicolor/`` 16x16,32x32,48x48,72x72,128x128,256x256 and 512x512 (which are displayed correctly), but none in 0x0. Bug?

#### Lock to launcher and two icons issue

I run the AppImage and it prompts for integration with desktop (install .desktop-file and icons), I confirm. I open the application and log in. The app itself works as it should (haven't tested it properly).

Whilst open, the application opens a second instance with Tutanota icon in my dock. If i choose "lock to launcher" (dock), the icon and desktop file from the current instance is copied to ``.local/share/``.

If I then close the app and try to launch Tutanota with my launcher (default Unity), which recognizes the .desktopfile, and it now opens two icons again, none of them with the Tutanota icon. 

Upon the second launch the icon ``tutanota-desktop.png`` now exists in ``.local/share/icons/`` and a new desktop file ``tutanota-desktop.desktop`` is created in addition to the ``appimagekit-tutanota-desktop.desktop`` file.

#### Desktop files / dock

Both of these opens in the dock (which shows active applications) when I open the application through my launcher. After a while, the first (non-active) disappears. When I close the app, the active one also disappears from the dock as it should.

Desktop file ``appimagekit-tutanota-desktop.desktop``:

```
[Desktop Entry]
Name=Tutanota Desktop
Comment=The desktop client for Tutanota, the secure e-mail service.
Exec="<path>/tutanota-desktop-linux.AppImage" %U
Terminal=false
Type=Application
Icon=appimagekit-tutanota-desktop
StartupWMClass=Tutanota
X-AppImage-Version=3.44.9.187
MimeType=x-scheme-handler/mailto;
Categories=Network;
X-AppImage-BuildId=1EfdXOqPBoV9lPB0F6n9kMqO8H9
X-Desktop-File-Install-Version=0.23
X-AppImage-Comment=Generated by /tmp/.mount_tutanoUmR9JF/AppRun
TryExec=<path>/tutanota-desktop-linux.AppImage
```

The second desktop file ``tutanota.desktop`` which is created after opening app again after using "lock to launcher":

```
[Desktop Entry]
Encoding=UTF-8
Version=1.0
Type=Application
Name=<email> - Tutanota
Icon=tutanota-desktop
Exec=/tmp/.mount_tutanoW1mSKy/tutanota-desktop
StartupNotify=false
StartupWMClass=tutanota-desktop
OnlyShowIn=Unity;
X-BAMFGenerated=true
```

#### Screenshots

The icon for the first desktop file in launcher:

![screenshot from 2018-12-21 18-50-07](https://user-images.githubusercontent.com/10927304/50356138-e2a8ac80-0548-11e9-95af-4287f6afea42.png)

The icon for the second desktop file in launcher: (non-executable)

![screenshot from 2018-12-21 19-05-14-2](https://user-images.githubusercontent.com/10927304/50356747-1b498580-054b-11e9-8c7c-2e1e3fdf64ef.png)

The dock:

![screenshot from 2018-12-21 19-23-10](https://user-images.githubusercontent.com/10927304/50357316-809e7600-054d-11e9-8f57-ee9cac436623.png) (at first launch)

![screenshot from 2018-12-21 18-51-01](https://user-images.githubusercontent.com/10927304/50357529-2225c780-054e-11e9-8656-6bf3246b543c.png) (at follwoing launches)

### Issues

- Two .desktop files if you try to lock the icon to launcher, where only the first (``appimagekit-tutanota-desktop``) is the intended file
  - The other one (``tutanota``) refers to executable in ``/tmp/.mount_tutano<id>`` which where path (id) changes upon each launch of the app while the desktop file is not upated. Looks like it shouldn't be copied to ``.local/share/applications`` at all since it only exists in the ``/tmp/.mount_tutano<id>`` directory which is created upon launch.
- appimage-icon is in 0x0 directory - not recognized
  - Should be in 512x512 based on its resolution. Copying it there makes the launcher icon display correctly.
- App opens as two icons in the dock. One closes first while app is running.
  - Should only be one icon displayed. This should be the desktop-file referring to the appimage-file as this is the one that is launched as user. Locking this to launcher is not an issue.

**Comment by snaggen** [comment]
StartupWMClass=tutanota-desktop
Setting this in the appimagekit desktop file fixes the launcher under Gnome.

**Comment by matsest** [comment]
Yes, seems like that does indeed solve the launcher issue under Unity as well. Thanks!

**Comment by ganthern** [comment]
Thanks for the great report, we'll look into it!

**Comment by matsest** [comment]
Great! I'm not too sure about the Electron framework specifications, but I think the logo file needs to contain a resolution in its filename, e.g. iconname-512x512.png to be recognized when building the Linux app. See [docs](https://www.electron.build/configuration/linux).

**Comment by saraneth0o** [comment]
I had been having issues with the icon not showing up in the menu and taskbar on Linux. Not sure if this is the same issue.

I found a way to add the icon after some troubleshooting though. In the '`appimagekit-tutanota-desktop.desktop`' file I changed the 'Icon=' field from '`Icon=appimagekit-tutanota-desktop`' to '`Icon=/home/username/.local/share/icons/hicolor/0x0/apps/appimagekit-tutanota-desktop.png`'.

That is the default icon that the AppImage makes but for some reason doesn't link to it properly
